docs/strategies/retry.md

```diff
@@ -105,3 +105,312 @@ new ResiliencePipelineBuilder().AddRetry(new RetryStrategyOptions
 | `UseJitter`        | False                                                                      | Allows adding jitter to retry delays.                                                    |
 | `DelayGenerator`   | `null`                                                                     | Used for generating custom delays for retries.                                           |
 | `OnRetry`          | `null`                                                                     | Action executed when retry occurs.                                                       |
+
+## Patterns and Anti-patterns
+Throughout the years many people have used Polly in so many different ways. Some reoccuring patterns are suboptimal. So, this section shows the donts and dos.
+
+### 1 - Overusing builder methods
+
+❌ DON'T
+
+Use more than one `Handle/HandleResult`
+```cs
+var retry = new ResiliencePipelineBuilder()
+    .AddRetry(new() {
+        ShouldHandle = new PredicateBuilder()
+        .Handle<HttpRequestException>()
+        .Handle<BrokenCircuitException>()
+        .Handle<TimeoutRejectedException>()
+        .Handle<SocketException>()
+        .Handle<RateLimitRejectedException>(),
+        MaxRetryAttempts = ...,
+    })
+    .Build();
+```
+**Reasoning**:
+- Even though this builder method signature is quite concise you repeat the same thing over and over (_please trigger retry if the to-be-decorated code throws XYZ exception_).
+- A better approach would be to tell  _please trigger retry if the to-be-decorated code throws one of the retriable exceptions_.
+
+✅ DO
+
+Use collections and simple predicate functions
+```cs
+ImmutableArray<Type> networkExceptions = new[] {
+    typeof(SocketException),
+    typeof(HttpRequestException),
+}.ToImmutableArray();
+
+ImmutableArray<Type> policyExceptions = new[] {
+    typeof(TimeoutRejectedException),
+    typeof(BrokenCircuitException),
+    typeof(RateLimitRejectedException),
+}.ToImmutableArray();
+
+var retry = new ResiliencePipelineBuilder()
+    .AddRetry(new() {
+        ShouldHandle = ex => new ValueTask<bool>(
+            networkExceptions.Union(policyExceptions).Contains(ex.GetType())),
```

Comment:
This is actually really slow as uses multiple ling extensions. Better idea is to cache the final array before and use it here.

Using switch expressions should be also pretty fast:

``` csharp
var retry = new ResiliencePipelineBuilder()
    .AddRetry(new() {
        ShouldHandle = args => args.Outcome.Exception switch
       HttpRequestException => PredicateResult.True(),
       BrokenCircuitException => PredicateResult.True(),
       // ...
    })
    .Build();
```

Reply:
I've refactored the code in a way that the `networkExceptions.Union(policyExceptions)` is pre-calculated rather than computed on-demand. I've called it `retryableExceptions`.

Reply:
I think we are good to merge then, please just sign the licence/cla agreement.

Reply:
@martintmk Can you please help me how to do that? The link inside the `contribute.md` redirects me to a pdf. How shall I proceed with that?

Reply:
I think for me it was some prompt in the PR UI that I just had to confirm. Isn't it showing to you?

Reply:
No unfortunately :(

Reply:
based on the docs this thing should work:

```
@microsoft-github-policy-service agree
```

Let me check with @martincostello once he comes back

Reply:
Can you try?

```
@dotnet-policy-service agree
```

Reply:
@martintmk It worked, thanks! 👍